core/thread/inc/ThreadLocalStorage.h

```diff
@@ -315,15 +168,7 @@ T &TTHREAD_TLS_INIT(ArgType arg) {
 
 #else // __cplusplus
 
-#if defined(R__HAS_THREAD_LOCAL)
```

Comment:
This removed section is the one that should stay (since now 'R__HAS_THREAD_LOCAL' is always true), right?

Reply:
and then `R__HAS_DECLSPEC_THREAD` can probably also be removed ...

Reply:
As far as I understand the code, those macros are for the case of C code:
https://github.com/root-project/root/blob/ac1741e49561b529670023ace6669029485abb50/core/thread/inc/ThreadLocalStorage.h#L67-L82

Reply:
(yes, the entire file is not very intuitive to follow; not sure if ROOT can require C11 so we can also use `thread_local` in the bit of C code left)

Reply:
> No, this section is for C code

I missed that :(

Reply:
So, it sound like instead, we ought to have a combination of
```
#if __STDC_VERSION__ >= 201112L
#define R_HAS_C_Thread_local
#endif
```
and 
```
# define TTHREAD_TLS_INIT(type,name,value) _Thread_local type name = (value)
```